File: Leetcode Weekly Contest Solution/1736. Latest Time by Replacing Hidden Digits.py

class Solution:
    
    def maximumTime(self, time):
        
        def check(tik, time):
            for char1, char2 in zip(tik, time):
                
                if char2 == '?' or char1 == char2: continue
                return False
            
            return True
            
            
        for i in range(23, -1, -1):
            for j in range(59, -1, -1):
                
                tik = str(i).zfill(2) + ':' + str(j).zfill(2)
                if check(tik, time): return tik
                
        return ''

# time complexity 23 * 60 * 5                

File: Leetcode Weekly Contest Solution/1736. Latest Time by Replacing Hidden Digits.cpp

class Solution {
	
public:
	bool check(string time, char str[]) {
		
		for (int i = 0; i < 5; i++) {
			if (time[i] == str[i] || time[i] == '?') continue;
			return false;
		}
		return true; 
	}
	
	string maximumTime(string time) {
		
		for (int i = 23; i >= 0; i--) {
			for (int j = 59; j >= 0; j--) {
				
				char str[20];
				sprintf(str, "%02d:%02d", i, j);
				if (check(time, str)) return str;
			}
		}
		
		return "";
	}
}